src/compiler/checker.ts

```diff
@@ -16089,6 +16089,21 @@ namespace ts {
             return false;
         }
 
+        function getJavaScriptClassType(symbol: Symbol): Type | undefined {
+            if (symbol && isDeclarationOfFunctionOrClassExpression(symbol)) {
+                symbol = getSymbolOfNode((<VariableDeclaration>symbol.valueDeclaration).initializer);
+            }
+            if (isJavaScriptConstructor(symbol.valueDeclaration)) {
+                return getInferredClassType(symbol);
+            }
+            if (symbol.flags & SymbolFlags.Variable) {

```

Comment:
The `const A =` case tests the from an external file, while the `const B =` case tests the use locally. Both cases are relevant. Are you talking about a third case where we import something re-exported from a third file?